### ISSUE TITLE
Windows: System.Windows.Forms assembly isn't found in CLI calls

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#19564]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Run the following from `cmd.exe` (or Windows PowerShell):

```powershell
pwsh -nop -c "Add-Type -Assembly System.Windows.Forms && [System.Windows.Forms.Form]"
```

Note that running the code in-session in a PowerShell Core session works fine.

### Expected behavior

```console
IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
True     False    Form                                     System.Windows.Forms.ContainerControl
```

That is, the assembly should load and make the `Form` type available.

### Actual behavior

```console
Add-Type: Cannot find path 'C:\Users\jdoe\System.Windows.Forms.dll' because it does not exist.
```

That is, it only looks for the assembly in the _current_ directory rather than in `$PSHOME` (which is what happens in-session).

### Environment data

```powershell
PowerShell Core 7.4.0-preview.3 on Windows 11 22H2
```

**Comment by 237dmitry** [comment]
> it only looks for the assembly in the current directory rather than in $PSHOME

Is this behavior only with this assembly? In 7.3.4 on windows works. On Linux in 7.4.0-preview.3 works with available (but not preloaded) assemblies. I tested `System.Linq.Parallel`

```
pwsh -nop -c { Add-Type -AssemblyName System.Linq.Parallel && [Linq.ParallelEnumerable] }
```

**Comment by mklement0** [comment]
My bad: 

I accidentally ran the v7.3.4 CLI on my ARM64 Windows 11, and in 7.3, due to an unrelated problem, the WinForms assemblies weren't included.

The latter problem was fixed in 7.4.0-preview.3, and everything works as intended.
